ios-target/View/Home/HomeViewController.swift

```diff
@@ -7,40 +7,80 @@
 //
 
 import UIKit
+import MapKit
 
 class HomeViewController: UIViewController {
   
   // MARK: - Outlets
   
-  @IBOutlet weak var welcomeLabel: UILabel!
-  @IBOutlet weak var logOut: UIButton!
+  @IBOutlet weak var mapView: MKMapView!
   
   var viewModel = HomeViewModel()
   
   // MARK: - Lifecycle Events
+  
   override func viewDidLoad() {
     super.viewDidLoad()
     viewModel.delegate = self
-    logOut.setRoundBorders(22)
+    setLocationServices()
+    viewModel.locationManager.checkLocationServices()
+    setLocationServices()
+    //TODO:
+//    viewModel.loadTargetPoints()
   }
   
   // MARK: - Actions
-  
-  @IBAction func tapOnGetMyProfile(_ sender: Any) {
-    viewModel.loadUserProfile()
-  }
 
-  @IBAction func tapOnLogOutButton(_ sender: Any) {
+  @IBAction func tapOnAddTarget(_ sender: Any) {
+    //targetFrom.showTargetForm()
+  }
+  
+  @IBAction func tapOnLogoutButton(_ sender: Any) {
     viewModel.logoutUser()
   }
+  
+  func setLocationServices() {
+    if CLLocationManager.locationServicesEnabled() {
+      viewModel.locationManager.setUpLocationManager()
+      checkLocationAuthorization()
+    }
+  }
+  
+  func checkLocationAuthorization() {
+    switch CLLocationManager.authorizationStatus() {
+    case .authorizedWhenInUse:
+      mapView.showsUserLocation = true
+      centerViewOnUserLocation()
+    case .denied:
+      //TODO:
+      //Show alert instricting them how to turn on permissions
+      break
+    case .notDetermined:
+      viewModel.locationManager.requestWhenInUseAuthorization()
+    case .restricted:
+      //TODO:
+      //tell them why you cant show the location
+      break
+    case .authorizedAlways:
+      break
+    }
+  }
+  
+  func centerViewOnUserLocation() {
+    if let location = viewModel.locationManager.locationManager.location?.coordinate {
```

Comment:
try to do as little business logic as possible on the view, so here for example, the viewModel could expose a `lastCoordinate` value so you don't need to drill down to the locationManager from here.